gremlin-python/src/main/python/gremlin_python/statics.py

```diff
@@ -74,12 +75,35 @@ def __init__(self, gremlin_type):
         self.gremlin_type = gremlin_type
 
 
-# TODO scale and unscaled value implementations are not following definition, need to revisit
 class BigDecimal(object):
+    """
+    Provides a way to represent a BigDecimal for Gremlin.
+    """
     def __init__(self, scale, unscaled_value):
         self.scale = scale
         self.unscaled_value = unscaled_value
 
+    @property
+    def value(self):
+        self._as_decimal = Decimal(self.unscaled_value)
+        precision = len(self._as_decimal.as_tuple().digits)
+        decimal.getcontext().prec = precision
```

Comment:
Does this cause the precision on a global level?

Reply:
Actually it does, I'll update to either reset or use local context.

Reply:
Updated with using a local context.